File: src/mainMot.py
#!/home/pi/vEnvs/rosPy/bin/python

'''
The main motor node. All motor nodes share identical structure but subscribe to
their respective topics and are defined by unique GPIO Pins.
'''

from rospy import loginfo, init_node, Subscriber, spin
from motors.msg import DOFArray
from motorControls import axisMotor

#home ang = -15.3943910599

mainMot = axisMotor((22, 23), 1, 60, 7.5)

def mainAngCall(data):
    '''
    Logs the angle calculated by the master and feeds it to the main motor.
    '''
    loginfo(data.mainAng)
    mainAng = data.mainAng
    mainMot.moveToAngle(mainAng)

def mainMotNode():
    '''
    Subscriber Node for the main motor.
    '''
    init_node('mainMotor', anonymous=True)
    Subscriber('motAngs', DOFArray, mainAngCall)
    spin()

if __name__ == '__main__':
    mainMot.gpioInit()
    mainMotNode()

File: src/nodes_cpp/mainMot.cpp
#include "ros/ros.h"
#include "motors/DOFArray.h"
#include "AxisMotor/AxisMotor.h"
#include <string>

#include <iostream>
using namespace std;

AxisMotor mainMot(22, 23, 1, 60, 7.5);

void callback(const motors::DOFArray::ConstPtr& msg)
{
    float str = msg->mainAng;
    ROS_INFO("[%s]\n", to_string(str));
    mainMot.moveToAngle(msg->mainAng);
}

int main(int argc, char **argv) {

    ros::init(argc, argv, "mainMot");
    ros::NodeHandle node;
    ros::Subscriber chat_top_sub = node.subscribe("motAngs", 1000, callback);
    ros::spin();

    return 0;
}